packages/components/Checkbox/src/Checkbox.tsx

```diff
@@ -63,7 +53,7 @@ export const Checkbox = compose<CheckboxType>({
       return (
         <Slots.root {...mergedProps}>
           {Checkbox.state.labelIsBefore && labelComponent}
-          <Slots.checkbox>
+          <Slots.checkbox {...mergedProps}>
```

Comment:
Passing in the entire merged props seems like it could cause some bugs. (What if a style prop was directly passed in, or a callback that is only meant to fire once?) I think it would be better to extract only the props you need to pass down here.

Reply:
Done

Reply:
Changes completed and approved , hence resolving.